project/apps/identity_provider/views/geoserver_integration.py

```diff
@@ -129,6 +129,14 @@ def introspect_api_key(self, api_key_uuid):
 
         if api_key.revoked:
             raise ValidationError(f"API key {api_key.key} is revoked.")
+        
+        # Check if the token has not a resource scope
+        if api_key.scope != 'resource':
```

Comment:
@Gpetrak as suggested by @giohappy i would consider to move the permissions checks inside a rest_framework permissions class as described in the [documetation](https://www.django-rest-framework.org/api-guide/permissions/#custom-permissions). You can find plenty example of the usage also in geonode